configs/DEMO_RISCV.yaml

```diff
@@ -0,0 +1,141 @@
+---
+# This file is based off of the current tx2.yaml config and serves as an example configuration for RISC-V cores.
+# The following resources where utilised to create the config file and naming schemes:
+# https://en.wikichip.org/wiki/cavium/microarchitectures/vulcan
+
+Core:
+  ISA: rv64
```

Comment:
On the name `rv64`; if `rv32` is supported, are there any major differences in the architecture from a SimEng point of view? i.e. would `rv32` and `rv64` require seperate Architecture.cc, Instruction.cc, etc classes?

If they will share these classes then would a ISA name of `RISCV` be more appropriate? Is there an explicit need to differentiate between `rv32` and `rv64`...

Reply:
rv32 and rv64 use some of the same opcodes but define instruction behaviour slightly differently (e.g. sign extensions) as they are defining differently sized registers. Even if the same arch specific files were to be used, there needs to be some way to differentiate between the two so SimEng knows how to set up the register file and how to interpret the instructions correctly. This information cannot be lost. Do you think there is a more appropriate way of passing that info in to SimEng?

Reply:
OK this makes sense then, thanks for clarifying 👍